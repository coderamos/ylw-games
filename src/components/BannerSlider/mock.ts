export const mock = [
  {
    image:
      'https://images.unsplash.com/photo-1496979149951-4fd28cf221e7?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=1024&ixid=eyJhcHBfaWQiOjF9&ixlib=rb-1.2.1&q=80&w=2048',
    title: 'Title example I',
    subtitle: '<p>Lorem ipsum <strong>dolor</strong> sit amet',
    buttonLabel: 'buy now',
    buttonLink: '/games/te1',
    ribbon: 'soon'
  },
  {
    image:
      'https://images.unsplash.com/photo-1496283529737-a39ef33f214d?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=1024&ixid=eyJhcHBfaWQiOjF9&ixlib=rb-1.2.1&q=80&w=2048',
    title: 'Title example II',
    subtitle: '<p>Lorem ipsum <strong>dolor</strong> sit amet',
    buttonLabel: 'buy now',
    buttonLink: '/games/te2'
  },
  {
    image:
      'https://images.unsplash.com/photo-1496979424495-8ca276b5bf34?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=1024&ixid=eyJhcHBfaWQiOjF9&ixlib=rb-1.2.1&q=80&w=2048',
    title: 'Title example III',
    subtitle: '<p>Lorem ipsum <strong>dolor</strong> sit amet',
    buttonLabel: 'buy now',
    buttonLink: '/games/te3'
  }
];
