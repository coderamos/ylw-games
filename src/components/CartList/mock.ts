export const mock = [
  {
    image:
      'https://www.rockstargames.com/reddeadredemption2/rockstar_games/r_d_r2_core/img/home_features/17.jpg',
    title: 'Red Dead Redemption II',
    price: 'R$ 159,00'
  },
  {
    image:
      'https://images.gog-statics.com/5643a7c831df452d29005caeca24c28cdbfaa6fbea5a9556b147ee26d325fa70_bg_crop_1680x655.jpg',
    title: 'Cyberpunk 2077',
    price: 'R$ 235,00'
  },
  {
    image:
      'https://images.gog-statics.com/7ca8104e8282031e4ea31e94af58ade233c5f83a48edc632fa7d85ffef320ac7_bg_crop_1680x655.jpg',
    title: 'Life is Strange',
    price: 'R$ 99,00'
  }
];
