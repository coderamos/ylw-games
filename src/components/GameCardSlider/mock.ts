export const mock = [
  {
    title: 'Cyberpunk 2077',
    developer: 'CD Projekt Red',
    image:
      'https://images.gog-statics.com/5643a7c831df452d29005caeca24c28cdbfaa6fbea5a9556b147ee26d325fa70_bg_crop_1680x655.jpg',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00'
  },
  {
    title: 'Red Dead Redemption II',
    developer: 'Rockstar Games',
    image:
      'https://www.rockstargames.com/reddeadredemption2/rockstar_games/r_d_r2_core/img/home_features/16.jpg',
    price: 'R$ 159,00',
    promotionalPrice: 'R$ 129,00'
  },
  {
    title: 'Detroit Become Human',
    developer: 'Quantic Dream',
    image:
      'https://cdn2.unrealengine.com/Diesel%2Fproductv2%2Fdetroit-become-human%2Fhome%2FDETROIT_BECOME_HUMAN_PC_SCREENSHOT_7-1920x1080-fad5546bf18b8ea5b408a3415605dfa29d4ec7d3.jpg',
    price: 'R$ 189,00',
    promotionalPrice: 'R$ 149,00'
  },
  {
    title: 'Life is Strange',
    developer: 'Square Enix',
    image:
      'https://images.gog-statics.com/7ca8104e8282031e4ea31e94af58ade233c5f83a48edc632fa7d85ffef320ac7_bg_crop_1680x655.jpg',
    price: 'R$ 99,00',
    promotionalPrice: 'R$ 49,00'
  },
  {
    title: 'Life is Strange: Before the Storm - Deluxe Edition',
    developer: 'Square Enix',
    image:
      'https://images.gog-statics.com/764a74f376251dba5a1624e3798fe7d55788dc579c604469fa76d1ae5fb22630_bg_crop_1680x655.jpg',
    price: 'R$ 199,00',
    promotionalPrice: 'R$ 99,00'
  },
  {
    title: 'Batman: The Enemy Within - The Telltale Series',
    developer: 'Athlon Games',
    image:
      'https://images.gog-statics.com/6afb33e28c42a647a180c9557418fa73e717d4c6fc9cda24cbeda9704eeff23e_bg_crop_1680x655.jpg',
    price: 'R$ 49,00',
    promotionalPrice: 'R$ 29,00'
  }
];
