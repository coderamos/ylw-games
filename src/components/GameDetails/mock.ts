export default {
  developer: 'CD PROJEKT RED',
  releaseDate: '2020-12-10T23:00:00',
  platforms: ['windows'],
  publisher: 'CD PROJEKT RED',
  rating: 'BR18',
  genres: ['action', 'role-playing']
};
