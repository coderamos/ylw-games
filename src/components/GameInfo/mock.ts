export const mock = {
  title: 'game title',
  price: '99,99',
  description:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi ad eveniet maxime cumque, excepturi nihil delectus tempore quaerat distinctio facere.'
};
