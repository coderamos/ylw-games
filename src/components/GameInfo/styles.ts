import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import * as RibbonStyles from 'components/Ribbon/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    background-color: ${theme.colors.white};
    padding: ${theme.spacings.xsmall};

    ${RibbonStyles.Wrapper} {
      right: -1rem;

      &:before {
        border-right-width: 1rem;
      }
    }

    ${media.greaterThan('medium')`
      ${RibbonStyles.Wrapper} {
        right: ${theme.spacings.small};
        top: ${theme.spacings.small};

        &:before {
          border: none;
        }
      }
    `}
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.grayWeb};
    margin-bottom: ${theme.spacings.small};
    text-align: justify;

    ${media.greaterThan('small')`
      max-width: 70%;
    `}
  `}
`;

export const ButtonsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;

    > button {
      width: 100%;
      margin-bottom: 0;

      & + button {
        color: ${theme.colors.secondary};
      }
    }

    ${media.greaterThan('medium')`
      flex-direction: row-reverse;

      > button {
        width: initial;
        margin-bottom: 0;
      }
    `}
  `}
`;
