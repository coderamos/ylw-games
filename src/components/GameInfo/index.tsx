import Button from 'components/Button';
import Heading from 'components/Heading';
import Ribbon from 'components/Ribbon';

import { ShoppingCart as ShoppingCartIcon } from '@styled-icons/feather';
import { Heart as HeartIcon } from '@styled-icons/feather';

import * as s from './styles';

export type GameInfoProps = {
  title: string;
  price: string;
  description: string;
};

const GameInfo = ({ title, description, price }: GameInfoProps) => (
  <s.Wrapper>
    <Heading color="secondary" lineBottom>
      {title}
    </Heading>
    <Ribbon color="secondary">${price}</Ribbon>
    <s.Description>{description}</s.Description>

    <s.ButtonsWrapper>
      <Button icon={<ShoppingCartIcon />}>add to cart</Button>
      <Button icon={<HeartIcon />} minimal>
        wishlist
      </Button>
    </s.ButtonsWrapper>
  </s.Wrapper>
);

export default GameInfo;
