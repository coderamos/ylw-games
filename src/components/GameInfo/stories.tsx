import { Story, Meta } from '@storybook/react/types-6-0';

import GameInfo, { GameInfoProps } from '.';
import { gameInfoMock as gameInfoArgs } from './mock';

export default {
  title: 'Design System/GameInfo',
  component: GameInfo,
  args: {
    ...gameInfoArgs
  },
  parameters: {
    backgrounds: {
      default: 'yellow'
    }
  },
  decorators: [
    Story => (
      <div style={{ margin: '0 auto', maxWidth: '130rem', padding: '2rem' }}>
        <Story />
      </div>
    )
  ]
} as Meta;

export const Default: Story<GameInfoProps> = args => <GameInfo {...args} />;
