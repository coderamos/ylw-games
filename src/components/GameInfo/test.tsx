import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import GameInfo from '.';

import { gameInfoMock as gameInfoProps } from './mock';

describe('<GameInfo />', () => {
  it('should render game info', () => {
    const { container } = renderWithTheme(<GameInfo {...gameInfoProps} />);

    expect(
      screen.getByRole('heading', { name: /game title/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/\$99,99/i)).toBeInTheDocument();
    expect(
      screen.getByText(
        /Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi ad eveniet maxime cumque, excepturi nihil delectus tempore quaerat distinctio facere/i
      )
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render buttons', () => {
    renderWithTheme(<GameInfo {...gameInfoProps} />);

    expect(
      screen.getByRole('button', { name: /add to cart/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /wishlist/i })
    ).toBeInTheDocument();
  });
});
