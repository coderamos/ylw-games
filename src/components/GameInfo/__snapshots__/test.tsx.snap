// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render game info 1`] = `
.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #FFCB00;
  border-radius: 0.4rem;
  border: none;
  color: #171717;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  height: 4rem;
  padding: 0.8rem 3.2rem;
}

.c6:hover {
  background: #FFBA00;
}

.c6 svg {
  width: 1.5rem;
}

.c6 svg + span {
  margin-left: 0.8rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #FFCB00;
  border-radius: 0.4rem;
  border: none;
  color: #171717;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  height: 4rem;
  padding: 0.8rem 3.2rem;
  background: none;
  color: #FFCB00;
}

.c8:hover {
  background: none;
}

.c8 svg {
  width: 1.5rem;
}

.c8 svg + span {
  margin-left: 0.8rem;
}

.c8:hover {
  color: #cca200;
}

.c1 {
  color: #171717;
  margin-bottom: 3.2rem;
  position: relative;
  font-size: 2.0rem;
}

.c1::after {
  position: absolute;
  content: '';
  border-bottom: 0.5rem solid #FFCB00;
  bottom: -1rem;
  left: 0;
  width: 5rem;
}

.c3 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FFFFFF;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
  right: -2rem;
  background-color: #171717;
  color: #FFCB00;
}

.c3::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c3::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c3::before {
  border-left-color: #000;
  border-top-color: #000;
}

.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  position: relative;
  background-color: #FFFFFF;
  padding: 1.6rem;
}

.c0 .c2 {
  right: -1rem;
}

.c0 .c2:before {
  border-right-width: 1rem;
}

.c4 {
  font-size: 1.4rem;
  color: #808080;
  margin-bottom: 2.4rem;
  text-align: justify;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > button {
  width: 100%;
  margin-bottom: 0;
}

.c5 > button + button {
  color: #171717;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c0 .c2 {
    right: 2.4rem;
    top: 2.4rem;
  }

  .c0 .c2:before {
    border: none;
  }
}

@media (min-width:450px) {
  .c4 {
    max-width: 70%;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .c5 > button {
    width: initial;
    margin-bottom: 0;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="secondary"
  >
    game title
  </h2>
  <div
    class="c2 c3"
    color="secondary"
  >
    $
    99,99
  </div>
  <p
    class="c4"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi ad eveniet maxime cumque, excepturi nihil delectus tempore quaerat distinctio facere.
  </p>
  <div
    class="c5"
  >
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="none"
        focusable="false"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="9"
          cy="21"
          r="1"
        />
        <circle
          cx="20"
          cy="21"
          r="1"
        />
        <path
          d="M1 1h4l2.68 13.39a2 2 0 002 1.61h9.72a2 2 0 002-1.61L23 6H6"
        />
      </svg>
      <span
        class=""
      >
        add to cart
      </span>
    </button>
    <button
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="none"
        focusable="false"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.84 4.61a5.5 5.5 0 00-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 00-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 000-7.78z"
        />
      </svg>
      <span
        class=""
      >
        wishlist
      </span>
    </button>
  </div>
</div>
`;
